build/gen_stub.php

```diff
@@ -854,6 +936,57 @@ function parseFunctionLike(
     );
 }
 
+function parseProperty(
+    Name $class,
+    PrettyPrinterAbstract $prettyPrinter,
+    int $flags,
+    Stmt\PropertyProperty $property,
+    ?Node $type
+): PropertyInfo {
+    $comment = $property->getDocComment();
+    $isDeprecated = false;
+    $isKnownName = false;
+    $docType = false;
+
+    if ($comment) {
+        $tags = parseDocComment($comment);
+        foreach ($tags as $tag) {
+            if ($tag->name === 'deprecated') {
+                $isDeprecated = true;
+            } else if ($tag->name === 'known') {
+                $isKnownName = true;
+            } else if ($tag->name === 'var') {
+                $docType = true;
+            }
+        }
+    }
+
+    $propertyType = $type ? Type::fromNode($type) : null;
+    if ($propertyType === null && !$docType) {
+        //throw new Exception("Missing type for property $class::\$$property->name");
+    }
+
+    if ($property->default instanceof Expr\ConstFetch &&
+        $property->default->name->toLowerString() === "null" &&
+        $propertyType && !$propertyType->isNullable()
+    ) {
+        $simpleType = $propertyType->tryToSimpleType();
+        if ($simpleType === null) {
+            throw new Exception(
+                "Property $class::\$$property->name has null default, but is not nullable");
```

Comment:
Hmm, maybe we could refactor the `FunctionOrMethodName` class (at least its name) a tiny little bit so that it becomes suitable for properties.